google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java

```diff
@@ -169,18 +169,25 @@ public Void apply(UnaryCallSettings.Builder<?, ?> builder) {
       // released.
       BatchingSettings oldBatchSettings =
           logBuilder.writeLogEntriesSettings().getBatchingSettings();
+
+      // The BatchingSettings from LoggingOptions should override
+      // ones provided in oldBatchSettings
+      BatchingSettings batchingSettings = options.getBatchingSettings();
+
       logBuilder
           .writeLogEntriesSettings()
           .setBatchingSettings(
-              oldBatchSettings
-                  .toBuilder()
-                  .setFlowControlSettings(
-                      oldBatchSettings
-                          .getFlowControlSettings()
-                          .toBuilder()
-                          .setLimitExceededBehavior(LimitExceededBehavior.Block)
-                          .build())
-                  .build());
+              batchingSettings != null
+                  ? batchingSettings
+                  : oldBatchSettings
```

Comment:
nit: I think this is an opportunity to refactor the code. Consider renaming `oldBatchSettings` to `defaultBatchSettings`.

Reply:
Sure, makes sense